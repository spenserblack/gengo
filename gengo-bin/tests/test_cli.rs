use insta::assert_snapshot;
use std::io::{self, Write};

const ROOT: &str = env!("CARGO_MANIFEST_DIR");

struct NullWriter;

impl Write for NullWriter {
    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
        Ok(buf.len())
    }
    fn flush(&mut self) -> io::Result<()> {
        Ok(())
    }
}

macro_rules! assert_stdout_snapshot {
    ($cli_args:expr $(,)?) => {{
        let cli = gengo_bin::cli::try_new_from($cli_args).unwrap();
        let mut stdout = Vec::new();
        let mut stderr = NullWriter;
        cli.run(&mut stdout, &mut stderr).unwrap();
        let stdout = String::from_utf8(stdout).unwrap();
        assert_snapshot!(stdout);
    }};
}

#[test]
fn test_javascript_repo() {
    assert_stdout_snapshot!(&["gengo", "-r", "test/javascript", "-R", ROOT]);
}

#[test]
#[cfg_attr(windows, ignore)]
fn test_breakdown_javascript_repo() {
    assert_stdout_snapshot!(&["gengo", "-r", "test/javascript", "-R", ROOT, "--breakdown"]);
}

//
// TODO Add test_javascript_repo_windows
