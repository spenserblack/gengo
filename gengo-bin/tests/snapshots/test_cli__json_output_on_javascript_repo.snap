---
source: gengo-bin/tests/test_cli.rs
expression: json
---
{
  "bin.js": {
    "detectable": true,
    "documentation": false,
    "generated": false,
    "language": {
      "category": "programming",
      "color": "#F0DC4E",
      "name": "JavaScript",
      "nerd_font_glyph": ""
    },
    "size": 28,
    "vendored": false
  },
  "dist/bin.js": {
    "detectable": true,
    "documentation": false,
    "generated": true,
    "language": {
      "category": "prose",
      "color": "#000000",
      "name": "Plain Text",
      "nerd_font_glyph": null
    },
    "size": 62,
    "vendored": false
  },
  "docs/index.html": {
    "detectable": false,
    "documentation": true,
    "generated": false,
    "language": {
      "category": "markup",
      "color": "#E96228",
      "name": "HTML",
      "nerd_font_glyph": ""
    },
    "size": 26,
    "vendored": false
  },
  "node_modules/my-dependency/index.js": {
    "detectable": false,
    "documentation": false,
    "generated": false,
    "language": {
      "category": "programming",
      "color": "#F0DC4E",
      "name": "JavaScript",
      "nerd_font_glyph": ""
    },
    "size": 29,
    "vendored": true
  },
  "src/bin.ts": {
    "detectable": true,
    "documentation": false,
    "generated": false,
    "language": {
      "category": "programming",
      "color": "#2F74C0",
      "name": "TypeScript",
      "nerd_font_glyph": ""
    },
    "size": 62,
    "vendored": false
  }
}
